src/codemodder/codemodder.py

```diff
@@ -29,18 +30,14 @@ def update_code(file_path, new_code):
 
 
 def apply_codemod_to_file(
-    execution_context: CodemodExecutionContext,
+    base_directory: Path,
     file_context,
     codemod_kls: CodemodExecutorWrapper,
     source_tree,
+    dry_run: bool = False,
 ):
-    name = codemod_kls.id
     wrapper = cst.MetadataWrapper(source_tree)
-    codemod = codemod_kls(
-        CodemodContext(wrapper=wrapper),
-        execution_context,
-        file_context,
-    )
+    codemod = codemod_kls(CodemodContext(wrapper=wrapper), file_context)
```

Comment:
maybe the diff is making it hard to read and `execution_context` is still getting passed but don't we need it in each codemod instance? Maybe its attrs aren't getting used right now but soon `repo_manager` will be, right?

Reply:
No, the execution context is no longer passed to individual codemods. I made this change because previously each codemod could potentially update `execution_context` which is not thread-safe. Anything we pass is going to need to go through `FileContext` now (which maybe needs a better name), so we can add `RepoManager` or just relevant fields like `python_versions` or `dependencies` or something.

Reply:
I would love to combine classes at some point so we don't confuse FileContext and ExecutionContext